modules/alb/main.tf

```diff
@@ -1,339 +1,334 @@
+  count = var.enable_https_listener ? 0 : 1
+
+  load_balancer_arn = aws_lb.alb.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "fixed-response"
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "404 Not Found"
+      status_code  = "404"
+    }
+  }
+}
+
+
+# -------------------------------------------
+# CREATE ALB HTTPS LISTENERS
+# -------------------------------------------
+
+data "aws_acm_certificate" "cert" {
+  # will only fetch a certificate if https is enabled
+  count = var.enable_https_listener ? 1 : 0
+
+  domain      = var.hosted_zone_name
+  types       = ["AMAZON_ISSUED"]
+  most_recent = true
+
+  depends_on = [
+    var.hosted_zone_name
+  ]
+}
+
+resource "aws_lb_listener" "redirect" {
+  # creates the http redirect listener when https is enabled
+  count = var.enable_https_listener ? 1 : 0
+
+  load_balancer_arn = aws_lb.alb.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+
+  depends_on = [
+    data.aws_acm_certificate.cert
+  ]
+}
+
+resource "aws_lb_listener" "https" {
+  # creates the https listener when https is enabled
+  count = var.enable_https_listener ? 1 : 0
+
+  load_balancer_arn = aws_lb.alb.arn
+  port              = "443"
+  protocol          = "HTTPS"
+
+  certificate_arn = data.aws_acm_certificate.cert[0].arn
+  ssl_policy      = "ELBSecurityPolicy-TLS13-1-2-2021-06"
+
+  default_action {
+    type = "fixed-response"
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "404 Not Found"
+      status_code  = "404"
+    }
+  }
+
+  depends_on = [
+    data.aws_acm_certificate.cert
+  ]
+}
+
+
+# -------------------------------------------
+# CREATE ALB SECURITY GROUP
+# -------------------------------------------
+
+resource "aws_security_group" "alb" {
+  name        = "${var.alb_name}-sg"
+  description = "Terraform managed security group for ${var.alb_name} ALB."
+
+  vpc_id = var.vpc_id
+}
+
+resource "aws_vpc_security_group_egress_rule" "alb" {
+  security_group_id = aws_security_group.alb.id
+  description       = "Allow all outbound traffic."
+
+  cidr_ipv4   = "0.0.0.0/0"
+  ip_protocol = "tcp"
+  from_port   = 0
+  to_port     = 65535
+}
+
+resource "aws_vpc_security_group_ingress_rule" "http" {
+  security_group_id = aws_security_group.alb.id
+  description       = "Allow HTTP traffic to the ALB from anywhere."
+
+  cidr_ipv4   = "0.0.0.0/0"
+  ip_protocol = "tcp"
+  from_port   = 80
+  to_port     = 80
+}
+
+resource "aws_vpc_security_group_ingress_rule" "https" {
+  count = var.enable_https_listener ? 1 : 0
+
+  security_group_id = aws_security_group.alb.id
+  description       = "Allow HTTPS traffic to the ALB from anywhere."
+
+  cidr_ipv4   = "0.0.0.0/0"
+  ip_protocol = "tcp"
+  from_port   = 443
```

Comment:
Shouldn't this include port 80? Our production environments are set up to forward traffic from port 80 to port 443

Reply:
Yes there is a redirect listener included. above